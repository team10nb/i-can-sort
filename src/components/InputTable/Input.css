.table{
    height: 40px;
    width: 186px;
}

.tr{
    height: 10px;
    width:185px;
}

.th{
    width: 125px;
}

.td{
    width: 60px;
}
