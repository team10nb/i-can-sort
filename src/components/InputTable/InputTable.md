
### **Developer**

Shiliang Chen, Yijie Lu

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**

Input table is used for store the lagal input, illegal input, and their result. It is the component of InputTutorial. 

###  **Import**

```html
import React, {useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
```

###  **Props**


| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
|legalShuffle      |() => void|                                    |click shuffle can get legal input|
|illlegalShuffle   |() => void|                                    |click shuffle can get illlegal input|
|inputLegalString  |string[]  |["2,1,6,10","9,12,8,3","3,2,17,10",]|the lagal string table used |
|inputIllegalString|string[]  |["2,a,6,10","9,*,&,3","3,#,&,10",]  |the illagal string table used |
|legalArrays       |number[][]|["2,1,6,10","9,12,8,3","3,2,17,10",]|the lagal array animation should operate |
|isCompleteIllegal |boolean[] |[false,false,false,]                |the signal when should the cross svg be visible |
|isCompleteLegal   |boolean[] |[false,false,false,]                |the signal when should the tick svg be visible |
|legalClick        |() => void|                                    |click create should let the animation display |
|illegalClick      |() => void|                                    |click create should let the animation display |


###  **Methods**

| Name | Props | Return | Description |
| ---- | ----- | ------ | ----------- |
|createData|str|    string     |     get the string from the string[]      |
|InputTable|props|    a input table     |     a table store the lagal input, illegal input, and their result.      |

###  **Demo**

```jsx
<InputTable
    inputLegalString={["2,1,6,10","9,12,8,3","3,2,17,10",]}
    inputIllegalString={["2,a,6,10","9,*,&,3","3,#,&,10",]}
    legalArrays={["2,1,6,10","9,12,8,3","3,2,17,10",]}
    isCompleteIllegal={[false,false,false,]}
    isCompleteLegal={[false,false,false,]}
/>
```
