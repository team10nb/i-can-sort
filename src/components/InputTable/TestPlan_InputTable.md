## Test Plan 
--------
<br>

#### ComponentName: InputTable

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should contain title & button |test whether titles and buttons are rendered properly| | Contain Legal Input, Illegal Input, Input and Result; Contain 4 buttons |
| should match InputTable Snapshot | Compare new snapshot with old one. |  | Snapshots are the same. |








