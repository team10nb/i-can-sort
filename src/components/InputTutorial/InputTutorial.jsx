import React, { useState } from "react";
import {
    makeStyles,
} from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import InputBox from "../InputTable/InputTable";
import ExplainationBox from "../ExplainationBox/ExplainationBox";
import { motion } from "framer-motion";
import ErrorIcon from "@material-ui/icons/Error";
import { random } from "lodash";

const spring = {
    type: "spring", // a framer motion type that simulates spring
    damping: 15, //Strength of opposing force. If set to 0, spring will oscillate indefinitely
    stiffness: 150, //Stiffness of the spring. Higher values will create more sudden movement. Set to 100 by default.
    mass: 0.1, // Mass of the moving object. Higher values will result in more lethargic movement
};

const useStyles = makeStyles((theme) => ({
    div: {
        display: "flex",
        width: 740,
        height: 530,
        // marginTop:630,
        // marginLeft:250,
        "& > *": {
            margin: theme.spacing(1),
        },
    },
    card: {
        // margintop: 10,
        background: "#F0F0F0",
        width: 550,
        height: 530,
        display: "grid",
        justifyContent: "center",
    },
    InputBox: {
        marginTop: 5,
        background: "#F0F0F0",
        width: 180,
        // height: 520,
        alignItems: "center",
    },
    bars: {
        listStyle: "none",
        padding: 0,
        margin: 0,
        marginBottom: 15,
        position: "relative",
        display: "flex",
        flexWrap: "wrap-reverse",
        justifyContent: "center",
    },
    bar: {
        listStyle: "none",
        padding: 0,
        margin: 0,
        boxShadow: "0px 0px 2px 2px #88888833",
        borderRadius: "10px",
        marginBottom: "10px",
        marginRight: "15px",
        marginLeft: "15px",
        width: "30px",
        height: "140px",
        fontSize: "0.1em",
        justifyContent: "center",
    },
    barNumber: {
        fontSize: "16px",
        marginTop: "-23px",
        textAlign: "center",
        fontWeight: "600",
    },
    switchAnimation: {
        display: "grid",
        justifyContent: "center",
        alignContent: "flex-end",
        marginTop: "100px",
        height: "300px",
        width: "300px",
    },
}));

export default function InputTutorial(props) {
    const classes = useStyles();

    // The maximum value of pairs
    const PAIR = 3;
    // The maximum value of input number
    const MAXNUMBER = 25;
    // The minimun value of input number
    const MINNUMBER = 1;
    // The maximum length of input array
    const LENGTH = 4;
    // It is used to clean timeouts to pause
    const [timeOutIds, setTimeOutIds] = useState([]);
    // Indicate whether a legal input demonstration complete
    const [isCompleteLegal, setIsCompleteLegal] = useState([
        false,
        false,
        false,
    ]);
    // Indicate whether a illegal input demonstration complete
    const [isCompleteIllegal, setIsCompleteIllegal] = useState([
        false,
        false,
        false,
    ]);
    // Indicate whether 
    const [isLegalPlaying, setIsLegalPlaying] = useState(true);
    // The legal string
    const [legalString, setLegalString] = useState([
        "2,1,6,10",
        "9,12,8,3",
        "3,2,17,10",
    ]);
    // The legal array
    const [legalArray, setLegalArray] = useState([
        [2, 1, 6, 10],
        [9, 12, 8, 3],
        [3, 2, 17, 10],
    ]);
    // The illegal string
    const [illegalString, setIllegalString] = useState([
        "2,a,6,10",
        "9,*,&,3",
        "3,#,&,10",
    ]);
    // The illegal array
    const [illegalArray, setIllegalArray] = useState([
        ["2", "a", "6", "10"],
        ["9", "*", "&", "3"],
        ["3", "#", "&", "10"],
    ]);
    // The bars displayed to visulise the numbers
    const [bars, setBars] = useState(legalArray[0]);

    // It is used to clean timeouts to pause the animation
    const clearTimeouts = () => {
        timeOutIds.forEach((timeoutId) => clearTimeout(timeoutId));
        setTimeOutIds([]);
    };

    // To shuffle the legal input array
    const legalShuffle = () => {
        let triple = [];
        let tripleArr = [];
        for (let j = 0; j < PAIR; j++) {
            let array = [];
            for (let i = 0; i < LENGTH; i++) {
                // const element = array[i];
                array.push(random(MINNUMBER, MAXNUMBER - 5));
            }
            triple.push(array.join(","));
            tripleArr.push(array);
            // setStr(triple.join(","));
        }
        setLegalArray(tripleArr);
        setLegalString(triple);
        setBars([0, 0, 0, 0]);

        setIsCompleteLegal([false, false, false]);
        setIsLegalPlaying(true);
    };

    // To shuffle illegal input array
    const illegalShuffle = () => {
        var characters =
            "!@#$%^&*ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
        var charactersLength = characters.length;
        let triple = [];
        let tripleArr = [];

        for (let j = 0; j < PAIR; j++) {
            let array = [];
            for (let i = 0; i < LENGTH; i++) {
                array.push(
                    characters.charAt(
                        Math.floor(Math.random() * charactersLength)
                    )
                );
            }
            if (array.join(",").match(/^[0-9]+((,)[0-9]+)*$/)) {
                j--;
                continue;
            }
            triple.push(array.join(","));
            tripleArr.push(array);
        }

        setIllegalString(triple);
        setIllegalArray(tripleArr);
        setBars([0, 0, 0, 0]);

        setIsCompleteIllegal([false, false, false]);
        setIsLegalPlaying(true);
    };

    // handler of play button of legal input
    const legalClick = () => {
        clearTimeouts();
        const legalTimeoutIds = [];
        setBars([0, 0, 0, 0]);
        setIsLegalPlaying(true);
        setIsCompleteLegal([false, false, false]);

        legalArray.forEach((item, i) => {
            let timeoutId = setTimeout(
                (item) => {
                    if (i === 0) {
                        setIsCompleteLegal([true, false, false]);
                    } else if (i === 1) {
                        setIsCompleteLegal([true, true, false]);
                    } else if (i === 2) {
                        setIsCompleteLegal([true, true, true]);
                    }
                    setBars(item.sort((a, b) => a - b));
                    console.log(item);
                },
                (i + 1) * 700, //time interval
                item
            );
            legalTimeoutIds.push(timeoutId);
        });

        // Clear timeouts upon completion
        let timeoutId = setTimeout(clearTimeouts, (legalArray.length + 1) * 700);
        legalTimeoutIds.push(timeoutId);
        setTimeOutIds(legalTimeoutIds);
    };

    // handler of play button of illegal input
    const illegalClick = () => {
        clearTimeouts();
        const illegalTimeoutIds = [];
        setIsLegalPlaying(false);
        setIsCompleteIllegal([false, false, false]);

        illegalString.forEach((item, i) => {
            let timeoutId = setTimeout(
                (item) => {
                    if (i === 0) {
                        setIsCompleteIllegal([true, false, false]);
                    } else if (i === 1) {
                        setIsCompleteIllegal([true, true, false]);
                    } else if (i === 2) {
                        setIsCompleteIllegal([true, true, true]);
                    }
                    setBars(item);
                },
                (i + 1) * 700, //time interval
                item
            );
            illegalTimeoutIds.push(timeoutId);
        });

        // Clear timeouts upon completion
        let timeoutId = setTimeout(clearTimeouts, (legalArray.length + 1) * 700);
        illegalTimeoutIds.push(timeoutId);
        setTimeOutIds(illegalTimeoutIds);
    };

    return (
        <div className={classes.div}>
            <Card className={classes.card}>
                <div className={classes.switchAnimation}>
                    <div className={classes.bars}>
                        {isLegalPlaying ? (
                            bars.map((value, key) => (
                                <motion.li
                                    key={key} // each bar's identification
                                    layout
                                    transition={spring}
                                    style={{
                                        backgroundColor: "#00BFFF",
                                        height: value * 10 + "px",
                                    }}
                                    className={classes.bar}
                                >
                                    <div className={classes.barNumber}>
                                        {value}
                                    </div>
                                </motion.li>
                            ))
                        ) : (
                            <div
                                style={{
                                    display: "grid",
                                    justifyContent: "center",
                                    alignContent: "flex-end",
                                }}
                            >
                                <ErrorIcon
                                    style={{ fontSize: 60, color: "red" }}
                                />
                            </div>
                        )}
                    </div>
                    {isLegalPlaying ? (
                        ""
                    ) : (
                        <span
                            style={{ fontWeight: "600", marginBottom: "8px" }}
                        >
                            Could not sort illegal input {bars.toString()}
                        </span>
                    )}
                    <ExplainationBox width="30" height={3}>
                        {isLegalPlaying ? bars.toString() : "Error"}
                    </ExplainationBox>
                </div>
            </Card>
            <InputBox
                // className = {classes.InputBox}
                legalShuffle={legalShuffle}
                illegalShuffle={illegalShuffle}
                inputLegalString={legalString}
                inputIllegalString={illegalString}
                legalArrays={legalArray}
                // illegalArrays={illegalArrays}
                isCompleteIllegal={isCompleteIllegal}
                isCompleteLegal={isCompleteLegal}
                legalClick={legalClick}
                illegalClick={illegalClick}
            />
        </div>
    );
}
