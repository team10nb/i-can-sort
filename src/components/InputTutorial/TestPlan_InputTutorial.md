## Test Plan 
--------
<br>

#### ComponentName: InputTutorial

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should call InputTable |test whether InputTable component is proprly called| | InputTable has been called |
| should match InputTutorial snapshot | Compare new snapshot with old one. |  | Snapshots are the same. |








