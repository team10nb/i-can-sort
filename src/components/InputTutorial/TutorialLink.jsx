/*
    Author: Yijie Lu
*/
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import HelpOutlineOutlinedIcon from '@material-ui/icons/HelpOutlineOutlined';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import RotateLeftIcon from '@material-ui/icons/RotateLeft';
import Backdrop from '@material-ui/core/Backdrop';
import logo0 from '../../Resource/inputExp0.png';
import logo1 from '../../Resource/inputExp1.png';
import logo2 from '../../Resource/inputExp2.jpg';
import logo3 from '../../Resource/inputExp3.png';
import logo4 from '../../Resource/inputExp4.jpg';
import {color} from '../../scenes/mainPages/Correctness';

  const useStyles = makeStyles((theme) => ({
    backdrop: {
      zIndex: theme.zIndex.drawer + 1,
      color: "#212121",
    },
    card: {
      maxWidth: 580,
      minHeight: 550,
      borderRadius:"30px", 
    },
    avatar: {
      "& > *": {
        marginTop: theme.spacing(1),
        marginLeft: theme.spacing(1),
      },
    },
    icon:{
        color: "#696969"
    },
    actionArea:{
      maxWidth: 395,
      maxHeight: 570,
    },
    button:{
      display: "flex",
        "& > *": {
            marginBottom: theme.spacing(2.5),
            marginLeft: theme.spacing(3),
        },
        "& > * + *": {
            marginLeft: theme.spacing(55),
        },
     },
     resetButton:{    
      "& > *": {
        marginBottom: theme.spacing(1),
        marginLeft: theme.spacing(3),
    },
     }
  }));
  
  

  

  function getStepContent(step) {
    // const classes = useStyles();
    switch (step) {
      case 0:
        return (
          <CardActionArea  width='395px' height='580px'>
            <img src={logo0} alt="logo0" width='587' heigh='300'  />
          </CardActionArea>

        );
      case 1:
        return (
          <CardActionArea  width='395px' height='580px'>
            <img src={logo1} alt="logo1"  width='587' heigh='300' />

          </CardActionArea>
        );
      case 2:
      return (
        <CardActionArea  width='395px' height='580px'>
          <img src={logo2} alt="logo2"  width='587' heigh='300' />
          <CardContent>
            <Typography variant="body2" color="textSecondary" component="p">
              By inputting a number from 1 to 7, the algorithm
              will output the week name from Sunday to Saturday.         
            </Typography>             
          </CardContent>
        </CardActionArea>
      );
      case 3:
        return (
          <CardActionArea  width='395px' height='580px'>
            <img src={logo3} alt="logo3"   width='587' heigh='300'/>
            
          </CardActionArea>
        );
      default:
        return 'Unknown step';
    }
  }


 
  export  default function Explanation() {
    const classes = useStyles();

    const [open, setOpen] = React.useState(false);
 
    const handleClickOpen = () => {
      setOpen(true);
    };

    const handleClose = () => {
      setOpen(false);
    };

    const [activeStep, setActiveStep] = React.useState(0);  

    const steps = 5;
    
    const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);     
    };

    const handleBack = () => {
      setActiveStep((prevActiveStep) => prevActiveStep - 1);
    };
    const handleReset = () => {
      setActiveStep(0);
    };

    const card = (
      // <Dialog onClose={handleClose} aria-labelledby="simple-dialog-title" open={open} classes = {classes.dialog}>
      //   <Fade in={open}>
      <Backdrop className={classes.backdrop} open={open} >
          <Card className={classes.card}>          
            <div className = {classes.avatar}>
                <IconButton aria-label="close"  onClick={handleClose} >
                  <HighlightOffIcon  style={{ color: color, opacity: 0.8, fontSize: 40  } } />
                </IconButton>
            </div> 
              {activeStep === steps - 1 ? (
                <div>
                  <div>
                <CardActionArea  width='395px' height='580px'>
                  <img src={logo4} alt="logo4"  width='587' heigh='300' />
                  <CardContent>
                    <Typography variant="body2" color="textSecondary" component="p">
                      For this page, the input set of this algorithm is positive intergers 1 to 7.&nbsp;Any other are illegal. This algorithm meets correctness since it can generate correct week name for any legal input.     
                    </Typography>             
                  </CardContent>
                </CardActionArea>    
              </div>
              <div className = {classes.resetButton}>
                <IconButton  onClick={handleReset} style={{ backgroundColor: color, opacity:0.5, }} size='small'><RotateLeftIcon style={{ color: 'white', fontSize:'40px'}}/></IconButton>          
              </div>
                </div>              
              ) : (
                <div>
                  <div>
                    {getStepContent(activeStep)}                    
                  </div>
                  <div className = {classes.button}>               
                  <IconButton  disabled={activeStep === 0} onClick={handleBack}  style={{ backgroundColor: color, opacity:0.5, }} size='small'><ChevronLeftIcon style={{ color: 'white', fontSize:'40px'}}/></IconButton>           
                  <IconButton  onClick={handleNext} style={{ backgroundColor: color, opacity:0.5, }} size='small'><ChevronRightIcon style={{ color: 'white', fontSize:'40px'}}/></IconButton>             
                  </div>
                </div>                
                )}
          </Card> 
      </Backdrop> 
    );

  
    return(
      <div style={{height:"48px"}}>
        <Tooltip title="tutorial" placement="bottom" arrow>
          <IconButton aria-label="help"  onClick={handleClickOpen}>
            <HelpOutlineOutlinedIcon className = {classes.icon}/>   
          </IconButton>
        </Tooltip>
        {card}
      </div>
    )
  }

