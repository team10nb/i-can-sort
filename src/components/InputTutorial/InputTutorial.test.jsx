/*
    Author: Shiliang Chen, Yijie Lu
*/

import { render } from "@testing-library/react";
import InputTutorial from "./InputTutorial";
import * as InputTables from "../InputTable/InputTable";

test( 'should call InputTable', () => {
    const inputTableSpy = jest
        .spyOn(InputTables, "default")
        .mockImplementation(() => <div>mockInputTable</div>);


    const content = render(<InputTutorial/>);

    expect(inputTableSpy).toBeCalled();

})

test( 'should match InputTutorial snapshot', () => {
    const inputTableSpy = jest
        .spyOn(InputTables, "default")
        .mockImplementation(() => <div>mockInputTable</div>);


    const content = render(<InputTutorial/>);

    expect(content).toMatchSnapshot();

})
