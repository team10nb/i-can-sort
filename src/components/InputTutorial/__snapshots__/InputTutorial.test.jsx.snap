// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match InputTutorial snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-div-19"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-20 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="makeStyles-algorithmTitle-21"
          >
            What day is (
            <span
              style="font-weight: 600; font-size: 20px; color: rgb(255, 111, 0);"
            >
              1
            </span>
            ) of a week?
          </div>
          <div
            class="makeStyles-display-24"
          >
            <div
              class="makeStyles-output-22"
            >
              <span
                style="font-weight: 600; font-size: 60px; color: rgb(255, 111, 0);"
              >
                Monday
              </span>
            </div>
            
            <div
              class="makeStyles-root-28"
            >
              <div
                class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-29 MuiPaper-outlined MuiPaper-rounded"
              >
                No.1 day of a week is Monday
              </div>
            </div>
          </div>
          <div
            class="makeStyles-divider-23"
          >
            <hr
              class="MuiDivider-root"
            />
            <div
              class="makeStyles-explanation-25"
            >
              <div
                style="height: 48px;"
              >
                <button
                  aria-label="help"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  title="tutorial"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-icon-33"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <div
                  aria-hidden="true"
                  class="MuiBackdrop-root makeStyles-backdrop-30"
                  style="opacity: 0; visibility: hidden;"
                >
                  <div
                    class="MuiPaper-root MuiCard-root makeStyles-card-31 MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="makeStyles-avatar-32"
                    >
                      <button
                        aria-label="close"
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            style="color: rgb(255, 111, 0); opacity: 0.8; font-size: 40px;"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiCardActionArea-root"
                          style="width: 580px; height: 415px; margin-bottom: 15px;"
                          tabindex="0"
                          type="button"
                        >
                          <img
                            alt="logo0"
                            heigh="300"
                            src="inputExp0.png"
                            width="587"
                          />
                          <span
                            class="MuiCardActionArea-focusHighlight"
                          />
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div
                        class="makeStyles-button-35"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                          disabled=""
                          style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              style="color: white; font-size: 40px;"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                              />
                            </svg>
                          </span>
                        </button>
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              style="color: white; font-size: 40px;"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-links-26"
              >
                <p
                  class="MuiTypography-root makeStyles-text-27 MuiTypography-body1"
                >
                  For any 
                  <span
                    style="color: rgb(255, 111, 0);"
                  >
                    legal input
                  </span>
                  , expected output can be generated
                </p>
              </div>
            </div>
          </div>
        </div>
        <div>
          mockInputTable
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-div-19"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-20 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="makeStyles-algorithmTitle-21"
        >
          What day is (
          <span
            style="font-weight: 600; font-size: 20px; color: rgb(255, 111, 0);"
          >
            1
          </span>
          ) of a week?
        </div>
        <div
          class="makeStyles-display-24"
        >
          <div
            class="makeStyles-output-22"
          >
            <span
              style="font-weight: 600; font-size: 60px; color: rgb(255, 111, 0);"
            >
              Monday
            </span>
          </div>
          
          <div
            class="makeStyles-root-28"
          >
            <div
              class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-29 MuiPaper-outlined MuiPaper-rounded"
            >
              No.1 day of a week is Monday
            </div>
          </div>
        </div>
        <div
          class="makeStyles-divider-23"
        >
          <hr
            class="MuiDivider-root"
          />
          <div
            class="makeStyles-explanation-25"
          >
            <div
              style="height: 48px;"
            >
              <button
                aria-label="help"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                title="tutorial"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-icon-33"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <div
                aria-hidden="true"
                class="MuiBackdrop-root makeStyles-backdrop-30"
                style="opacity: 0; visibility: hidden;"
              >
                <div
                  class="MuiPaper-root MuiCard-root makeStyles-card-31 MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="makeStyles-avatar-32"
                  >
                    <button
                      aria-label="close"
                      class="MuiButtonBase-root MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          style="color: rgb(255, 111, 0); opacity: 0.8; font-size: 40px;"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div>
                    <div>
                      <button
                        class="MuiButtonBase-root MuiCardActionArea-root"
                        style="width: 580px; height: 415px; margin-bottom: 15px;"
                        tabindex="0"
                        type="button"
                      >
                        <img
                          alt="logo0"
                          heigh="300"
                          src="inputExp0.png"
                          width="587"
                        />
                        <span
                          class="MuiCardActionArea-focusHighlight"
                        />
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="makeStyles-button-35"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                        disabled=""
                        style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            style="color: white; font-size: 40px;"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                            />
                          </svg>
                        </span>
                      </button>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                        style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            style="color: white; font-size: 40px;"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-links-26"
            >
              <p
                class="MuiTypography-root makeStyles-text-27 MuiTypography-body1"
              >
                For any 
                <span
                  style="color: rgb(255, 111, 0);"
                >
                  legal input
                </span>
                , expected output can be generated
              </p>
            </div>
          </div>
        </div>
      </div>
      <div>
        mockInputTable
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
